### PR TITLE
Trying to improve the replication scheduler configuration parameters (fixes apache/couchdb#344)

### PR DESCRIPTION
**Pull request by skeyby**
## Overview

Detailed a bit more the replication scheduler parameters.

## Testing recommendations

The changes are in section 3.9, replication configuration.

## GitHub issue number

This pull requests fixes issue #344 I previously opened.

## Checklist

- [X] Documentation is written and is accurate;
- [X] `make check` passes with no errors
- [ ] Update [rebar.config.script]

### CODE REVIEW

File: src/config/replicator.rst [hunk 1]
```diff
@@ -27,11 +27,18 @@ Replicator Database Configuration
 
         .. versionadded:: 2.1
 
-        Number of actively running replications. Making this too high could
-        cause performance issues. Making it too low could mean replications
-        jobs might not have enough time to make progress before getting
-        unscheduled again. This parameter can be adjusted at runtime and will
-        take effect during next rescheduling cycle::
+        Number of actively running replications.
+        This value represent the threshold to trigger the automatic replication
```
Comment (flimzy) [inline src/config/replicator.rst]:
```suggestion
        This value represents the threshold to trigger the automatic replication
```

File: src/config/replicator.rst [hunk 2]
```diff
@@ -27,11 +27,18 @@ Replicator Database Configuration
...
+        scheduler.
+        The system will check every ``interval`` milliseconds how many replication
+        jobs are running, and if there are more than ``max_jobs`` active jobs,
+        the scheduler will pause-and-restart up to ``max_churn`` jobs in the
+        scheduler queue.
+        Making this  valute too high could cause performance issues, while making
```
Comment (flimzy) [inline src/config/replicator.rst]:
```suggestion
        Making this value too high could cause performance issues, while making
```

File: src/config/replicator.rst [hunk 3]
```diff
@@ -40,8 +47,9 @@ Replicator Database Configuration
 
         .. versionadded:: 2.1
 
-        Scheduling interval in milliseconds. During each reschedule cycle
-        scheduler might start or stop up to "max_churn" number of jobs::
+        Scheduling interval in milliseconds.
+        During each reschedule cycle scheduler might start or stop up to ``max_churn``
```
Comment (flimzy) [inline src/config/replicator.rst]:
```suggestion
        During each reschedule cycle the scheduler might start or stop up to ``max_churn``
```

File: src/config/replicator.rst [hunk 4]
```diff
@@ -50,10 +58,10 @@ Replicator Database Configuration
 
         .. versionadded:: 2.1
 
-        Maximum number of replications to start and stop during rescheduling.
-        This parameter along with ``interval`` defines the rate of job
-        replacement. During startup, however a much larger number of jobs could
-        be started (up to ``max_jobs``) in a short period of time::
+        Maximum number of replication jobs to start and stop during rescheduling.
+        This parameter along with ``interval`` defines the rate of job replacement.
```
Comment (flimzy) [inline src/config/replicator.rst]:
```suggestion
        This parameter, along with ``interval``, defines the rate of job replacement.
```

File: src/config/replicator.rst [hunk 5]
```diff
@@ -50,10 +58,10 @@ Replicator Database Configuration
...
+        During startup, however a much larger number of jobs could be started
```
Comment (flimzy) [inline src/config/replicator.rst]:
```suggestion
        During startup, however, a much larger number of jobs could be started
```

**Comment by skeyby** [comment]
Thanks for the corrections. I'll double read better next time...

**Comment by wohali** [comment]
@nickva As the author of the new replication scheduler, can you please review and +1 this PR before we merge it?

**Comment by nickva** [comment]
+1

Thanks for your contribution, @skeyby

**Pull request merged**